notebooks/sycamore-tutorial-intermediate-etl.ipynb

```diff
@@ -0,0 +1,410 @@
+    "\n",
+    "## Partitioning data\n",
+    "\n",
+    "We can now continue with our ETL pipeline. The next step is to partition the DocSet, which chunks each document into smaller elements using labeled bounding boxes. Each element could be a titles, table, table row and column, image, regular text, and more. This also extracts the actual information from the element (e.g. the information from the table), or does OCR. We use the [Aryn Partitioning Service (APS)](https://sycamore.readthedocs.io/en/stable/aryn_cloud/aryn_partitioning_service.html) to do this, and you can [get a free API key here](https://www.aryn.ai/get-started).\n",
+    "\n",
+    "In this example, we've set APS to extract table structures using a table structure recognition model, and to use OCR to get text from the document (but not to extract image contents). Using OCR takes a bit more time, but it can yield slightly higher quality text in some cases."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sycamore.transforms.partition import ArynPartitioner\n",
+    "# Set your Aryn API key. See https://sycamore.readthedocs.io/en/stable/aryn_cloud/accessing_the_partitioning_service.html#using-sycamore-s-partition-transform\n",
+    "\n",
+    "partitioned_docset = (\n",
+    "        initial_docset.partition(partitioner=ArynPartitioner(extract_images=False,  extract_table_structure=True, use_ocr=True))\n",
+    "        .materialize(path=\"./pc-tutorial/partitioned\", source_mode=sycamore.materialize_config.MaterializeSourceMode.IF_PRESENT)\n",
+    ")\n",
+    "partitioned_docset.execute()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We make a remote call to APS for each document. APS will take a few minutes to partition these documents, as there are a few hundred pages across the PDF doucments. \n",
+    "\n",
+    "Notice that we include a `materialize` operation and force execution with `execute`. `materialize` writes the DocSet out to disk, and it is configured to be used as a checkpoint in the pipeline. Subsequent operations will read from this materialized DocSet. In this notebook, we will use commands like `show_pages` to visualize the DocSet for educational purposes (a production pipeline wouldn't do this), and these commands force recomputation of the DocSet. If present, Sycamore will use the materialized DocSet instead of recomputing it (and avoiding running the APS step again)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can visualize the pages of the PDF using the `show_pages` function, so you can see how APS chunked each page into elements.\n",
+    "\n",
+    "Note that this requires the `poppler` library to be installed on your system. You can install it on a Mac with `brew install poppler`, or in Debian with `apt-get install poppler-utils`."
```

Comment:
I'd suggest changing `Mac` to `macOS` and Debian to `Linux` to not be overly specific

Reply:
Ack